xray/commands/audit/sca/yarn/yarn.go

```diff
@@ -1,16 +1,40 @@
 package yarn
 
 import (
+	"errors"
+	"fmt"
+	"github.com/jfrog/build-info-go/build"
 	biUtils "github.com/jfrog/build-info-go/build/utils"
+	"github.com/jfrog/gofrog/version"
+	rtutils "github.com/jfrog/jfrog-cli-core/v2/artifactory/commands/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/artifactory/commands/yarn"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/config"
 	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
 	"github.com/jfrog/jfrog-cli-core/v2/xray/commands/audit/sca"
 	"github.com/jfrog/jfrog-cli-core/v2/xray/utils"
 	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
 	"github.com/jfrog/jfrog-client-go/utils/log"
 	xrayUtils "github.com/jfrog/jfrog-client-go/xray/services/utils"
+	"path/filepath"
 )
 
-func BuildDependencyTree() (dependencyTrees []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
+const (
+	// Do not execute any scripts defined in the project package.json and its dependencies.
+	v1IgnoreScriptsFlag = "--ignore-scripts"
+	// Run yarn install without printing installation log.
+	v1SilentFlag = "--silent"
+	// Disable interactive prompts, like when there’s an invalid version of a dependency.
+	v1NonInteractiveFlag = "--non-interactive"
+	// Skips linking and fetch only packages that are missing from yarn.lock file
+	v2UpdateLockfileFlag = "--mode=update-lockfile"
+	// Ignores any build scripts
+	v2SkipBuildFlag     = "--mode=skip-build"
+	yarnV2Version       = "2.0.0"
+	nodeModulesRepoName = "node_modules"
+)
+
+func BuildDependencyTree(params utils.AuditParams) (dependencyTrees []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
```

Comment:
Structs pass by value, make this change in other functions if needed
```suggestion
func BuildDependencyTree(params *utils.AuditParams) (dependencyTrees []*xrayUtils.GraphNode, uniqueDeps []string, err error) {
```